src/core/control/GeometryToolController.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Xournal++
+ *
+ * A geometry tool controller
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <gtk/gtk.h>
+
+#include "gui/PageView.h"
+#include "model/GeometryTool.h"
+#include "util/Point.h"
+
+class Stroke;
+
+/**
+ * @brief A class that controls a geometry tool
+
+ * The geometry tool can be moved, rotated and scaled
+ * There are methods for translating coordinates
+ * and methods to deal with the temporary stroke
+ */
+
+class GeometryToolController {
+public:
+    GeometryToolController(XojPageView* view, GeometryTool* geometryTool);
+    ~GeometryToolController();
+
+public:
+    /**
+     * @brief moves the geometry tool in x- and y-direction
+     * @param x the translation in x-direction (in document coordinates)
+     * @param y the translation in y-direction (in document coordinates)
+     */
+    void move(double x, double y);
+
+    /**
+     * @brief rotates the geometry tool around its rotation center
+     * @param da the rotation angle
+     * @param cx the x-coordinate of the rotation center (in document coordinates)
+     * @param cy the y-coordinate of the rotation center (in document coordinates)
+     */
+    void rotate(double da, double cx, double cy);
+
+    /**
+     * @brief resizes the geometry tool by the factor f with respect to a given scaling center
+     * @param f the scaling factor
+     * @param cx the x-coordinate of the scaling center (in document coordinates)
+     * @param cy the y-coordinate of the scaling center (in document coordiantes)
+     */
+    void scale(double f, double cx, double cy);
+
+    /**
+     * @brief marks a point with a "x"
```

Comment:
A picture says more than words:
![grafik](https://user-images.githubusercontent.com/40485433/205459403-55a76a0d-6979-4289-bc42-893f1b6790b0.png)
Here the setsquare origin is marked. It can be useful for geometric constructions to mark the rotation center, even more so for the compass in the future.

Reply:
Hopefully it's clear now.